### ISSUE TITLE
Want to show whole numbers, graph shows fractions. 

### ISSUE THREAD

**Issue by SimonAM** [indiespirit/react-native-chart-kit#544]
I'm trying to display whole numbers or counting numbers. The graph shows fractions. How do i handle this?

Example: 
![image](https://user-images.githubusercontent.com/3391354/128815443-d968e474-a158-4d0b-97c6-a973c90e1fee.png)

**Comment by OAK-02** [comment]
`chartConfig={
           decimalPlaces: 0,
}`

**Comment by SimonAM** [comment]
Leads to 5 segments with 0, 0, 1, 1, 1 respectively.

**Comment by nfmshow** [comment]
@SimonAM, please were you able to find a solution?

**Comment by SimonAM** [comment]
You can set the number of segments dynamically, but the max value and step size between segments are calculated automatically.  
Here are similar issues:
#235
#122
#70

There is a proposed solution, #253. I have not tested it yet though.

**Comment by SimonAM** [comment]
I can confirm that #253  does let you specify ylabels. 

A caveat is that your maxvalue will always be place on your maximum ylabel. You cant have your maximum datapoint at 97 and your maximum ylabel at 100 without 97 ending up at the 100 segment line.  

This is a typescript solution, using a mix of:

-  ramda   
- fp-ts
 - fp-ts-std
 - fp-ts-ramda. 

Where the Dataset type is imported from react-native-chart-kit. 

``` 
 const highestDatapoint = (datasets: Dataset[]) => {
    const maxOfArray = R.apply(Math.max);
    const arrayOfMaxes = A.map(flow(
      prop("data"),
      maxOfArray,
    ))(datasets)
    return maxOfArray(arrayOfMaxes);

  };



  const maxYVal = flow(
    highestDatapoint,
    R.flip(R.divide)(10),
    Math.ceil,
    R.multiply(10)
  )

  const stepSize = (maxval: number): number =>
    guard([
      [R.lt(1000), R.always(1000)],
      [R.lt(500), R.always(100)],
      [R.lt(250), R.always(50)],
      [R.lt(100), R.always(25)],
      [R.lt(50), R.always(10)],
      [R.lt(10), R.always(5)],
      [R.lt(0), R.always(1)],
    ])(R.always(1))(maxval)


  //const getNSteps = maxval => maxval / getStepsize(maxval);

  /*   const createYLabels = (maxvalue) => R.pipe(
      getNSteps,
      R.inc,
      R.range(0),
      R.map(R.multiply(getStepsize(maxvalue))),
      R.map(Math.ceil)
    )(maxvalue) */

  const createYLabs = (maxval: number) => (segmentsize: number) => (n: number) =>
    n > maxval
      ? false
      : n % segmentsize === 0
        ? [n, n + 1]
        : ["", n + 1]


  /*
  * Generator helper function that allows us to
  * cleanly return a label from an array
  * of labels for each segment of the y-axis
  */
  function* yLabel() {
    yield* R.pipe(
      maxYVal,
      stepSize,
      createYLabs(maxYVal(datasets)),
      R.flip(R.unfold)(0)
    )(datasets)

  }

  const yLabelIterator = yLabel();
  
  <LineChart 
      {...props}
       formatYLabel={() => yLabelIterator.next().value}
      segments={highestDatapoint(datasets)}
      withVerticalLines={false}
      withHorizontalLines={true}
  />
```


Since the dots have an offset, one could argue the graph isn't exact enough to need all the horizontal lines. However, although there is a yAxisInterval prop, no xAxisInterval prop is yet available (see #370 ). 

![yaxisinterval1](https://user-images.githubusercontent.com/3391354/150173230-cd5a1fa6-bf08-404e-b823-9f97710817da.png)
This is the result of a recording with the value 21. It's quite hard to see the value. Solutions that comes to mind is to change the style of every nth horizontal line, however cant see any solutions for that. A similar issue is open #497.
